### PR TITLE
Ashley/subscriber dashboard

### PR DESCRIPTION
**Pull request by ashleynguyen27**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?
Added a subscriber dashboard that can be used for subscribers and subscriber owners. This dashboard includes a billing, project news, and charts (not finished in this pr) sections. 

### How to Review
Check the SubscriberDashboard.js file.
This file was too big so it was split into sections. The name of the components are: 
- DashboardBillingSection
- DashboardChartsSection
- DashboardProjectNewsSection. 
Billing.js and Navbar.js were also modified. 

Demo: Log into sam@berkeley.edu to see an example subscriber dashboard. 

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

- SubscriberDashboard.js
- DashboardBillingSection
- DashboardChartsSection
- DashboardProjectNewsSection 
- Billing.js
- Navbar.js

Demo: Log into sam@berkeley.edu to see an example subscriber dashboard. The Subscriber board should look like the prototype Iris made 

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Tests Performed, Edge Cases

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[
Screen Shot 2020-03-12 at 9 28 46 PM
](url)

CC: @dfangshuo

### CODE REVIEW

File: src/screens/subscriber/Billing.js [hunk 1]
```diff
@@ -55,27 +55,26 @@ class Billing extends React.Component {
 
   render() {
     const { mode, transactions, isReady, pendingBills } = this.state;
-    const { isLoadingUserData } = this.props;
+    const { isLoadingUserData, location } = this.props;
+    const { mode2 } = location;
```
Comment (aivantg) [inline src/screens/subscriber/Billing.js]: I see what you did here and it's pretty sweet, but I think a slightly cleaner way to do it, is instead of having two `mode` parameters and comparing based on both of them, you can keep the mode in a query string! 

So when you link to the billing page, instead of linking to `/billing`, link to `/billing?view=all`

and then in the constructor, you can pull out the value of `view` and set the initial state for `mode` based on that. I just recently added a library to help do that called `qs`. 

```
// Top of file
import qs from 'qs';

// In Constructor
const {location} = this.props
const view = qs.parse(location.search, {
  ignoreQueryPrefix: true
}).view;

if (view === 'all') { 
  this.state.mode = 1 // It's okay to edit state directly because this code would be in the constructor
}
```

This is actually important because otherwise, you would never be able to navigate away from the AllBillsview

File: src/components/NavBar.js [hunk 2]
```diff
@@ -47,7 +47,14 @@ class NavBar extends React.PureComponent {
                     pathname === '/billing' ? 'navItemSelected' : 'navItem'
                   }
                 >
-                  <Link to="/billing">Billing</Link>
+                  <Link
+                    to={{
+                      pathname: '/billing',
+                      mode2: 0
```
Comment (aivantg) [inline src/components/NavBar.js]: following with comment below, remove mode2 and this edit, instead use query parameters

File: src/screens/subscriber/SubscriberDashboard.js [hunk 3]
```diff
@@ -1,9 +1,159 @@
...
+    this.setState({
+      mode: 0
+    });
+  }
+
+  switchChartView() {
+    const { activeTab, hasShares } = this.state;
+    if (activeTab === 'My Solar Project' && hasShares) {
+      this.setState({
+        activeTab: 'Community Solar Projects'
```
Comment (aivantg) [inline src/screens/subscriber/SubscriberDashboard.js]: I def am not perfect about this either, but i think these should be constants. since the names are so long, we're one typo away from a very hard to find bug LOL

File: src/screens/subscriber/components/DashboardBillingSection.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  render() {
+    const { totalBalance, data } = this.props;
+    return (
+      <div>
+        <div className="subscriber-section-header">
+          <div className="subscriber-header">Billing Summary</div>
+          <Link
+            to={{
+              pathname: '/billing',
+              mode2: 0
```
Comment (aivantg) [inline src/screens/subscriber/components/DashboardBillingSection.js]: I think this one should be just `/billing` (with query param update)

File: src/screens/subscriber/components/DashboardBillingSection.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                    type="button"
+                    className={`subscriber-billing-make-payment-button ${
+                      totalBalance === 0 ? 'disabled' : ''
+                    }`}
+                    disabled={totalBalance === 0}
+                  >
+                    <Link
+                      to={{
+                        pathname: '/billing',
+                        mode2: 1
```
Comment (aivantg) [inline src/screens/subscriber/components/DashboardBillingSection.js]: here we should pass in the query param `/billing?view=all`

File: src/screens/subscriber/components/DashboardProjectNewsSection.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          <div className="subscriber-header">Project News</div>
+          <Link to="/projectnews">
+            <img src={RightArrow} alt="right arrow" />
+          </Link>
+        </div>
+        <div className="subscriber-side-section-body">
+          {isLoadingAnnouncements ? (
+            <LoadingComponent />
+          ) : (
+            announcements.map(announcement => {
```
Comment (aivantg) [inline src/screens/subscriber/components/DashboardProjectNewsSection.js]: I think ideally we make some sort of a reusable component across the many times that community cards are displayed...but this can be something we talk about in cleanup rather than something that falls under _this_ PR. Note for @dfangshuo

File: src/styles/SubscriberDashboard.css [hunk 7]
```diff
@@ -0,0 +1,184 @@
+.subscriber-page {
+  margin: 1rem 7rem;
```
Comment (niwong) [inline src/styles/SubscriberDashboard.css]: Ohhh man, should I be using rems? Do we have a standard for that?

File: src/screens/subscriber/Billing.js (hunk 1 above)
Comment (niwong) [reply src/screens/subscriber/Billing.js]: Oh that's a super cool idea.

File: src/styles/SubscriberDashboard.css (hunk 7 above)
Comment (ashleynguyen27) [reply src/styles/SubscriberDashboard.css]: tbh idk what is standard but i started using rem in the beginnning. i regret no one else uses it

**Comment by dfangshuo** [comment]
@ashleynguyen27  Follow the chronology of the Project News page (aka latest announcements should be on top). It should be a simple reverse list function (see how project news does it)

File: src/screens/subscriber/components/DashboardProjectNewsSection.js (hunk 6 above)
Comment (ashleynguyen27) [reply src/screens/subscriber/components/DashboardProjectNewsSection.js]: ya, I was actually going to delete in the meantime cause I hated seeing it for the millisecond it was on the screen.

File: src/screens/subscriber/components/DashboardBillingSection.js (hunk 4 above)
Comment (ashleynguyen27) [reply src/screens/subscriber/components/DashboardBillingSection.js]: ohhhhh tyty

File: src/screens/subscriber/components/DashboardChartsSection.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      activeTab: 0
+    };
+    this.switchChartView = this.switchChartView.bind(this);
+  }
+
+  // need to make sure the value doesnt switch if its the same button
+  switchChartView = e => {
+    const { activeTab } = this.state;
+    const { hasShares } = this.props;
+    if (e.target.value != 0 && activeTab === 0 && hasShares) {
```
Comment (dfangshuo) [inline src/screens/subscriber/components/DashboardChartsSection.js]: if you want to check that `e.target.value` is not null, just do 
```suggestion
    if (e.target.value && activeTab === 0 && hasShares) {
```

!= should be avoided as much as possible

File: src/screens/subscriber/components/DashboardChartsSection.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+      this.setState({
+        activeTab: 1
+      });
+    } else if (e.target.value != 1) {
```
Comment (dfangshuo) [inline src/screens/subscriber/components/DashboardChartsSection.js]: similarly: 
```suggestion
    } else if (e.target.value) {
```

**Comment by dfangshuo** [comment]
Use 
```
switchChartView = e => {
    const { hasShares } = this.props;
    if (!hasShares) {
      return;
    }

    // const { activeTab } = this.state;
    const toSelect = parseInt(e.target.value);

    this.setState(prevState => {
      const { activeTab: prevActiveTab } = prevState;
      
      if (prevActiveTab === toSelect) {
        return {};
      }

      return { activeTab : toSelect };
    });
  };

```

for the `!=` part
